packages/lib/action/service.ts

```diff
@@ -175,59 +176,77 @@ export const getActionsByEnvironmentId = async (environmentId: string, page?: nu
 };
 
 export const createAction = async (data: TActionInput): Promise<TAction> => {
-  validateInputs([data, ZActionInput]);
+  try {
+    validateInputs([data, ZActionInput]);
+    const { environmentId, name, properties, sessionId } = data;
 
-  const { environmentId, name, properties, sessionId } = data;
-
-  let eventType: TActionClassType = "code";
-  if (name === "Exit Intent (Desktop)" || name === "50% Scroll") {
-    eventType = "automatic";
-  }
+    let eventType: TActionClassType = "code";
+    if (name === "Exit Intent (Desktop)" || name === "50% Scroll") {
+      eventType = "automatic";
+    }
 
-  const session = await getSession(sessionId);
+    const session = await getSession(sessionId);
 
-  if (!session) {
-    throw new ResourceNotFoundError("Session", sessionId);
-  }
+    if (!session) {
+      throw new ResourceNotFoundError("Session", sessionId);
+    }
 
-  let actionClass = await getActionClassByEnvironmentIdAndName(environmentId, name);
+    let actionClass = await getActionClassByEnvironmentIdAndName(environmentId, name);
 
-  if (!actionClass) {
-    actionClass = await createActionClass(environmentId, {
-      name,
-      type: eventType,
-      environmentId,
-    });
-  }
+    if (!actionClass) {
+      actionClass = await createActionClass(environmentId, {
+        name,
+        type: eventType,
+        environmentId,
+      });
+    }
 
-  const action = await prisma.event.create({
-    data: {
-      properties,
-      session: {
-        connect: {
-          id: sessionId,
+    const action = await prisma.event.create({
+      data: {
+        properties,
+        session: {
+          connect: {
+            id: sessionId,
+          },
         },
-      },
-      eventClass: {
-        connect: {
-          id: actionClass.id,
+        eventClass: {
+          connect: {
+            id: actionClass.id,
+          },
         },
       },
-    },
-  });
+    });
+
+    const environmentPrisma = await prisma.environment.findUnique({
```

Comment:
@Dhruwang the create action function is currently called around 15k times an hour and we are just starting to scale the usage. This would introduce an extra database call for every action creation, even though it's only needed for the first action sent to Formbricks. I think we need to find another solution.
One option would be to at least get the environment cached using the service `getEnvironment`, but this still gives one extra database currently every 30 minutes (cache expires currently after 30 minutes). But it's still an option we can consider :-) 
Or we need to find a good central place in the formbricks admin interface where we set this.